### ISSUE TITLE
Is it possible to limit environment deployments to a specific `base_ref`?

### ISSUE THREAD

**Issue by redoz** [github/branch-deploy#193]
### Details

If I want to limit deployments to environment `A` and `B` to PRs against `main`, and deployments to environment `C` to PRs against `release/*`. Is that possible using this action? I figured I'd include a `github.base_ref` check and have two workflows, one for `A` & `B`, and one for `C`, but there's no `base_ref` in an `issue_comment` it seems.

**Comment by GrantBirki** [comment]
:wave: Hello again! Thanks for opening another issue and I would be happy to help.

I read over your issue a few times and my :brain: is getting a bit jumbled around with all the `a`, `b`, and `c` environments.

Do you possibly have a demo, or could you walk me through (in a comment) how this might work in the real world? I am just struggling to understand your message and could use some help visualizing your use case.

Thanks!

**Comment by redoz** [comment]
Yeah even as I wrote the question, I realized it was a bit confusing.  In effect, we have a few different environments, and we use branch-deploy to deploy to those from pull-requests. 
I would like to limit what environments are available for deploying onto, based on the base_ref of the pull request. If you PR targets `main`, you can use environment A, and B, but if your PR targets `release/*` you can only deploy to environment C

I would like to be able to write something like this:
```yaml
    name: 🚀 Deploy
    if: ${{ github.event.issue.pull_request  && github.base_ref == 'main' }}
    runs-on: ubuntu-latest
    steps:
      - name: Branch deployment
        id: branch-deploy
        uses: github/branch-deploy@v7.3.0
```
Then I could simply use two different workflows,  but I can't check the `base_ref` because this is an `issue_comment` not a `pull_request` trigger, so I was wondering if there was any kind of built-in support for this? I hope that makes more sense :)

**Comment by GrantBirki** [comment]
@redoz Thank you! That makes a whole lot more sense now. Currently, there isn't any logic in this Action that would support what you are after.

This would likely be some custom logic that you would need to bake into your workflow. You would need likely need to do the following in your workflow:

1. Make an API call to fetch the pull request data on which the `issue_comment` was left on
2. Check the `base.ref` of the pull request and also parse the comment manually to see what `environment` it is targeting
3. Conditional stop / continue your workflow to the branch deploy step if it meets your given criteria

In an effort to kinda help I opened a new PR so that the `base_ref` value is now an output of this Action. However, I'm not sure how much that will actually help you here since if you get to the part where you are processing the outputs of this Action, a deployment has likely already started.

You are in the custom / advanced deployment territory here and I wish I could help more!

**Comment by GrantBirki** [comment]
I accidentally clicked "close" instead of comment. Please let me know if there is anything else I can help with here

**Comment by redoz** [comment]
@GrantBirki  thanks for taking the time to get back to me. But indeed my plan was quite similar:
> 1. Make two workflows, one for `A` & `B` and one for `C`
> 2. Make an API call to fetch the pull request data on which the issue_comment was left on
> 3. Check the base.ref of the pull request 
> 4. Conditional stop / continue your workflow to the branch deploy step if it meets your given criteria

So effectively the same, but since I don't want to have to reimplement all `.deploy [[to] [env]` syntax parsing I was just going to split it into two separate workflows. 

But I figured before I went down that road, perhaps there was some way of configuring this action to do this for me so I thought I would ask :) I already use the `gh` tool in the pipeline for other purposes so it shouldn't be too hard to do the check myself, thanks again for your help, hope you have a great weekend!
